### ISSUE TITLE
phpcs-only sniffs still reported as fixable by phpcbf

### ISSUE THREAD

**Issue by Majkl578** [squizlabs/PHP_CodeSniffer#2008]
Fixable PHPCS-only sniffs are still reported as fixable by PHPCBF.

With this config:
```xml
<?xml version="1.0"?>
<ruleset>
    <rule ref="PSR2.ControlStructures.ElseIfDeclaration" phpcs-only="true"/>
</ruleset>
```
and this code:
```php
<?php

if (true) {
} else if (false) {
}
```

The report looks like:
```
FILE: /tmp/phpcs-bug/test.php
-----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
-----------------------------------------------------------------------
 4 | WARNING | [x] Usage of ELSE IF is discouraged; use ELSEIF instead
-----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------

Time: 14ms; Memory: 4Mb
```

This is not correct. PHPCBF cannot fix this sniff automatically.

Attempting to do so has no effect for the reported sniff, and in this case (since it's the only sniff registered) it eventually ends with error so it's even more obvious:
```
ERROR: No sniffs were registered

Run "phpcbf --help" for usage information
```

**Comment by gsherwood** [comment]
> This is not correct. PHPCBF cannot fix this sniff automatically.

That's not entirely true. PHPCBF can fix the error, but your ruleset has banned that. Using another ruleset that doesn't ban this would allow the error to be fixed.

But I know what you are getting at. I just don't know if that's something that I want to change because it's a little more complicated than just sniffs being registered or not. There may also be file paths, ignore statements, and other settings marked as phpcs-only (or even phpcbf-only) so I can't see how I can accurately decide if PHPCBF will fix a particular error without running PHPCS twice over every file (once normally, once using PHPCBF rules). That's not an acceptable performance trade-off for most people.

**Comment by Majkl578** [comment]
> PHPCBF can fix the error, but your ruleset has banned that

Yes, so it can't fix the error. :)

> Using another ruleset that doesn't ban this would allow the error to be fixed.

Following the same logic, you could say "_PHPCBF can fix errors for <some not included sniff>_".

My point here is that running `vendor/bin/phpcs` is reporting the error (based on auto-discovered config) and since it's marked as phpcs-only, it should not say it's fixable, since it's not fixable with the given ruleset by just running `vendor/bin/phpcbf`.

Here is my use case: https://github.com/doctrine/coding-standard/blob/643d2a11158a0cbf7f260a1c04e6eea2bdb15982/lib/Doctrine/ruleset.xml#L156-L157
It's a sniff that reports usage of `?:` that is able to fix the code by converting `$foo ?: $bar` to  `$[oo ? $foo : $bar`, but that is not desired since the intention is to ban the short variant, not require long one.

I think for now I'll work around that with a _$fixable_ property on the sniff itself.

**Comment by gsherwood** [comment]
I've had another think about this and I can't see how it could be implemented. By the time an error is generated, there is no way to connect the error to any part of the ruleset to determine if the error would have been generated during a PHPCBF run as well.

Closing as I don't see a way forward. It will remain a limitation of the report.
